test/extensions/filters/listener/original_dst/original_dst_fuzz_test.cc

```diff
@@ -0,0 +1,72 @@
+#include "envoy/extensions/filters/listener/original_dst/v3/original_dst.pb.h"
+
+#include "common/network/utility.h"
+
+#include "extensions/filters/listener/original_dst/original_dst.h"
+
+#include "test/extensions/filters/listener/original_dst/original_dst_fuzz_test.pb.validate.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/network/mocks.h"
+
+#include "gmock/gmock.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace OriginalDst {
+
+DEFINE_PROTO_FUZZER(
+    const envoy::extensions::filters::listener::original_dst::v3::OriginalDstTestCase& input) {
+
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  }
+
+  NiceMock<Network::MockListenerFilterCallbacks> callbacks_;
+  NiceMock<Network::MockConnectionSocket> socket_;
+
+  try {
+    auto address_ = Network::Utility::resolveUrl(input.address());
+    ON_CALL(socket_, localAddress()).WillByDefault(testing::ReturnRef(address_));
+    ON_CALL(callbacks_, socket()).WillByDefault(testing::ReturnRef(socket_));
+
+    if (address_ != nullptr) {
+      ON_CALL(socket_, addressType()).WillByDefault(testing::Return(address_->type()));
+      if (socket_.addressType() == Network::Address::Type::Ip) {
+        ON_CALL(socket_, ipVersion()).WillByDefault(testing::Return(address_->ip()->version()));
+      }
+    }
+
+    auto filter = std::make_unique<OriginalDstFilter>();
+
+    // Set address family of mock socket so that it routes correctly thru addressFromSockAddr
+    ON_CALL(socket_, getSocketOption(_, _, _, _))
+        .WillByDefault(testing::WithArgs<0, 2>(Invoke([](int level, void* optval) {
+          switch (level) {
+          case SOL_IPV6:
+            static_cast<sockaddr_storage*>(optval)->ss_family = AF_INET6;
+            break;
+          case SOL_IP:
+            static_cast<sockaddr_storage*>(optval)->ss_family = AF_INET;
+            break;
+          default:
+            NOT_REACHED_GCOVR_EXCL_LINE;
+          }
+
+          return Api::SysCallIntResult{0, 0};
+        })));
+
+    filter->onAccept(callbacks_);
+  } catch (const EnvoyException& e) {
+    ENVOY_LOG_MISC(debug, "EnvoyException: {}", e.what());
```

Comment:
What parts of this do we expect might throw an EnvoyException? Can we make the scope of this tighter?

Reply:
Yeah, only resolveUrl() and onAccept() can actually throw an EnvoyException. Fixed in the latest commit - let me know if it's good.

I was having scope issues before with declaring address_ (needed to be inside of the try-catch block in order to use auto) and then trying to access it later in the lambda to check which address family to use, but I have since changed the switch statement condition.

Reply:
Wait a minute... `onAccept()` throws?

Reply:
No, I've never seen it actually throw and I don't think it should... it doesn't REALLY need to be in a try-catch block.

Reply:
Perfect! We should definitely report throws from `onAccept` as crashes. There's an agenda to remove exceptions from the data plane, and this would constitute a larger part of that https://github.com/envoyproxy/envoy/issues/10878

Reply:
Bump on crashing on exceptions from `onAccept`?

Reply:
I moved it back into the try-catch block because of scope issues - if outside, the destructor for the FakeConnectionSocket is called before `onAccept`, which removes its function implementations from the vtable and causes a crash due to a pure virtual function being called. I can't create the FakeConnectionSocket without the address from `resolveUrl`, but if there's a cleaner way to do it, lmk!

Reply:
Would making it a pointer work?

Reply:
I see 2 fairly simple options:

1. declare `address` outside the `try` block (where it will be a null shared_ptr), set it inside the try, then construct `socket` and `filter` after the `catch`.
2. make `socket` a `std::unique_ptr<FakeConnectionSocket>` that is declared outside the block and initialize it inside. Then do the onAccept after the `catch`.

Reply:
I did the first option and it seems to work! The only line in the try-catch block now is `resolveUrl`, which is the only function that should throw an exception.